webapp/channels/src/packages/mattermost-redux/src/actions/profiles_statuses_groups.ts

```diff
@@ -0,0 +1,170 @@
+
+            dispatch(getProfilesByIds(Array.from(pendingUserIdsForProfiles)));
+            pendingUserIdsForProfiles.clear();
+        }
+
+        // Process immediately if the pending user ids exceeds the limit
+        if (pendingUserIdsForProfiles.size >= MAX_USER_PROFILES_BATCH) {
+            console.log('addUserIdForProfiles', 'executing immediately for', pendingUserIdsForProfiles.size);
+
+            getPendingProfilesById();
+        }
+
+        pendingUserIdsForProfiles.add(userId);
+
+        // Start the interval if it is not already running
+        if (userProfilesIntervalId === null) {
+            userProfilesIntervalId = setInterval(() => {
+                if (pendingUserIdsForProfiles.size > 0) {
+                    console.log('addUserIdForProfiles', 'executing with interval for pending', pendingUserIdsForProfiles.size);
+
+                    getPendingProfilesById();
+                } else {
+                    console.log('addUserIdForProfiles', 'no pending user ids');
+                }
+            }, USER_PROFILES_DURATION);
+        }
+
+        return {data: true};
+    };
+}
+
+export function cleanupUserProfilesInterval() {
+    if (userProfilesIntervalId !== null) {
+        clearInterval(userProfilesIntervalId);
+        userProfilesIntervalId = null;
+    }
+}
+
+function addUserIdForStatuses(userId: string): ActionFunc<boolean> {
+    return (dispatch) => {
+        function getPendingStatusesById() {
+            console.log('addUserIdForStatuses start', pendingUserIdsForStatuses);
+
+            // Since we can only fetch 100 user statuses at a time, we need to batch the requests
+            if (pendingUserIdsForStatuses.size >= MAX_USER_STATUSES_BUFFER) {
+                // We use temp buffer here to store up until max buffer size
+                // and clear out processed user ids
+                const userIds: string[] = [];
+
+                let counter = 0;
+                for (const pendingUserId of pendingUserIdsForStatuses) {
+                    userIds.push(pendingUserId);
+                    pendingUserIdsForStatuses.delete(pendingUserId);
+
+                    counter++;
+
+                    if (counter >= MAX_USER_STATUSES_BUFFER) {
+                        break;
+                    }
+                }
+
+                console.log('addUserIdForStatuses', 'executing for', userIds.length, '>');
+                dispatch(getStatusesByIds(userIds));
+            } else {
+                // If we have less than max buffer size, we can directly fetch the statuses
+                console.log('addUserIdForStatuses', 'executing for', pendingUserIdsForStatuses.size, '<');
+                dispatch(getStatusesByIds(Array.from(pendingUserIdsForStatuses)));
+                pendingUserIdsForStatuses.clear();
+            }
+
+            if (pendingUserIdsForStatuses.size >= USER_STATUSES_REQUEST_SURGE_THRESHOLD) {
```

Comment:
As discussed, we can probably remove this if we don't ever see it happening.

Reply:
removed